consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftRound.java

```diff
@@ -115,7 +118,28 @@ public ConsensusRoundIdentifier getRoundIdentifier() {
    * @param headerTimeStampSeconds the header time stamp seconds
    */
   public void createAndSendProposalMessage(final long headerTimeStampSeconds) {
-    final Block block = blockCreator.createBlock(headerTimeStampSeconds).getBlock();
+    final Block block;
+
+    final Optional<ScheduledProtocolSpec.Hardfork> shanghaiFork =
```

Comment:
Think a better approach might be to disable the withdrawals functionality for QBFT/IBFT.

You could do that by overriding the protocol schedule in BaseBftProtocolScheduleBuilder in the applyBftChanges method and configuring the protocol schedule to not use a withdrawalsProcessor and use the ProhibitedWithdrawals for the withdrawals validator i.e. same as before Shanghai.

Reply:
Yeah that's a good suggestion. It does seem cleaner to avoid validating a feature that QBFT doesn't support, rather than provide the withdrawal field so that validation passes. I'll possibly start a different draft PR that uses the alternative approach you've suggested (along with your other comments below) but I'll see how I go.